soundbay/utils/logging.py

```diff
@@ -110,18 +110,25 @@ def update_predictions(self, pred_tuple):
         self.pred_proba_list.append(torch.softmax(pred_tuple[0].data, 1).cpu().numpy())
         self.label_list += label.cpu().numpy().tolist()
 
-    def calc_metrics(self, epoch: int, mode: str = 'train', label_names: List[str] = ('Noise', 'Call')):
+    def calc_metrics(self, epoch: int, mode: str = 'train', label_names: List[str] = None):
         """calculates metrics, saves to tensorboard log & flush prediction list"""
         pred_proba_array = np.concatenate(self.pred_proba_list)
         self.metrics_dict = self.get_metrics_dict(self.label_list, self.pred_list, pred_proba_array)
-        self.log_writer.log({f'Metrics/{mode}_Accuracy': self.metrics_dict['accuracy']}, step=epoch)
-        self.log_writer.log({f'Metrics/{mode}_f1score': self.metrics_dict['f1score']}, step=epoch)
-        self.log_writer.log({f'Metrics/{mode}_precision': self.metrics_dict['precision']}, step=epoch)
-        self.log_writer.log({f'Metrics/{mode}_recall': self.metrics_dict['recall']}, step=epoch)
-        self.log_writer.log({f'Metrics/{mode}_auc': self.metrics_dict['auc']}, step=epoch)
+        for metric, value in self.metrics_dict['global'].items():
+            self.log_writer.log({f'Global Metrics {mode}/{metric}': value}, step=epoch)
+
+        if label_names is None:
```

Comment:
You can't do it much earlier because you don't know how many classes you have until you finished one epoch.
You shouldn't do it (IMO) above calc_metrics because then you'll have to do it once before every call to calc_metrics.